docbook/reference/en/en-US/modules/RESTEasy_Client_Framework.xml

```diff
@@ -234,67 +234,188 @@ public interface LibraryService {
+                   <row>
+                       <entry>org.jboss.resteasy.client.jaxrs.engines.ApacheHttpClient4Engine</entry>
+                       <entry>Uses HttpComponents Apache pre-4.3 api</entry>
+                   </row>
+                   <row>
+                       <entry>org.jboss.resteasy.test.profiling.InMemoryClientEngine</entry>
+                       <entry>Dispatches requests to a server in the same JVM</entry>
+                   </row>
+                   <row>
+                       <entry>org.jboss.resteasy.client.jaxrs.engines.URLConnectionEngine</entry>
+                       <entry>Uses java.net.HttpURLConnection</entry>
+                   </row>
+               </tbody>
+           </tgroup>
+       </table>
+
+       <para>
+           The Resteasy Client Framework can also be customized.  The user can provide
+           their own implementations of
+           <code class="classname">ClientHttpEngine</code> to the
+           <code class="classname">ResteasyClient</code>.
+       </para>
+
+       <programlisting>
+ClientHttpEngine myEngine = new ClientHttpEngine() {
+    protected SSLContext sslContext;
+    protected HostnameVerifier hostnameVerifier;
+
+    @Override
+    public ClientResponse invoke(ClientInvocation request) {
+        // implement your processing code and return a
+        // org.jboss.resteasy.client.jaxrs.internal.ClientResponse
+        // object.
+    }
+
+    @Override
+    public SSLContext getSslContext() {
+       return sslContext;
+    }
+
+    @Override
+    public HostnameVerifier getHostnameVerifier() {
+       return hostnameVerifier;
+    }
+
+    @Override
+    public void close() {
+       // do nothing
+    }
+};
+
+ResteasyClient client = new ResteasyClientBuilder().httpEngine(myEngine).build();
+       </programlisting>
+
+       <para>
+           Resteasy and <classname>HttpClient</classname> make reasonable default decisions so
+           that it is possible to use the client framework without ever referencing
+           <classname>HttpClient</classname>. For some applications it may be necessary to drill
+           down into the <classname>HttpClient</classname> details.
+           <classname>ApacheHttpClient43Engine</classname> and
+           <classname>ApacheHttpClient4Engine</classname> can
+           be supplied with an instance of
+           <classname>org.apache.http.client.HttpClient</classname> and an instance of
+           <classname>org.apache.http.protocol.HttpContext</classname>, which can carry
+           additional configuration details into the <classname>HttpClient</classname> layer.
+
+       </para>
+           The existing class constructor in <classname>ApacheHttpClient43Engine</classname> and
+           <classname>ApacheHttpClient4Engine</classname> that accepts input arguments,
+           <classname>HttpClient</classname>
+           and <classname>HttpContext</classname> has been deprecated.  It has been

```

Comment:
Is it really needed to explicitly mention currently deprecated methods/constructors? (really just asking if we can keep this simpler and provide deep details of not-deprecated facilities).